packages/intrepid2/unit-test/MonolithicExecutable/ProjectedGeometryTests.cpp

```diff
@@ -0,0 +1,293 @@
+// @HEADER
+// ************************************************************************
```

Comment:
Agreed.  I have just added two tests that check p convergence of curvilinear geometry to a circle, by doing:
1. Project circular geometry onto basis nodes.
2. Construct higher-order geometry object using nodal values.
3. Compute sum of the cell measures; compare with exact area of the circle to determine error.
4. Check that error for p is less than error for p-1.

The first test does this for a single element; the second test does it for a 2x2-element mesh.  Each goes from p=1 to MaxOrder; MaxOrder is 8 right now.  I am a little surprised the error is not smaller for the higher-order elements; here's the sequence for the 2x2 mesh:
```
error for p = 1: 0.0816278
error for p = 2: 0.0815687
error for p = 3: 0.0448938
error for p = 4: 0.0384696
error for p = 5: 0.0272936
error for p = 6: 0.0242242
error for p = 7: 0.0190003
error for p = 8: 0.0172987
```
Do those numbers seem plausible to you?  If you like, I can interrogate this a bit further.

In consequence, we also exercise the higher-order constructor of CellGeometry.  I did change the interface there to make it more convenient to use the result of ProjectedGeometry; I also had to make a couple other changes within CellGeometry to get the tests passing.

Reply:
@CamelliaDPG thanks for doing this! I'm also a bit surprised. Is this the map from square to a circle? If so it might be OK becasue the map is not smooth, right?

Reply:
@mperego, yes, that's correct; it's the map from a square to a circle, and I think you're probably right that it's the fact that the transformation is not smooth that explains the slow convergence in p.  The above was for 4x4 (I originally wrote 2x2, but that was incorrect).  Here are the numbers for 1 element:
```
error for p = 1: 1.14159
error for p = 2: 1.11215
error for p = 3: 0.624522
error for p = 4: 0.568132
error for p = 5: 0.408723
error for p = 6: 0.365485
error for p = 7: 0.285525
error for p = 8: 0.259215
```

And here's 8x8:
```
error for p = 1: 0.0207112
error for p = 2: 0.0207099
error for p = 3: 0.0113912
error for p = 4: 0.00969285
error for p = 5: 0.00686607
error for p = 6: 0.00608875
error for p = 7: 0.00477771
error for p = 8: 0.00435071
```
I do tend to think that it's probably OK, since it is converging.  Hard to think of a bug that would explain slow convergence here.  We could think about a test that involves some sort of smooth transformation instead, for sometime after this PR is in.

Reply:
Agreed! Thanks again for adding this test.

Reply:
Sure thing!